src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -0,0 +1,160 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * AribaConnector Class
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(ResourceConstants.PLUGIN_NAME)
+@Description("Connection to access data from Ariba.")
+
+public class AribaConnector implements Connector {
+  private static final Logger LOG = LoggerFactory.getLogger(AribaConnector.class);
+  private static final String METADATA_PATH = "api/analytics-reporting-view/v1";
+  private static final String PRODUCT = "product";
+  private static final String ANALYTICS = "analytics";
+  private static final String REALM = "realm";
+  private static final String ENTITY_TYPE_TEMPLATE = "template";
+  private static final Gson gson = new Gson();
+  private final AribaConnectorConfig config;
+  public String accessToken;
+
+  AribaConnector(AribaConnectorConfig config) {
+
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    try {
+      config.validateToken(collector);
+    } catch (AribaException | IOException e) {
+      LOG.error("Credentials are incorrect", e);
+    }
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    JsonArray jsonArray = listTemplates();
+    for (int i = 0; i < jsonArray.size(); i++) {
+      JsonObject resultJsonObject = gson.fromJson(jsonArray.get(i), JsonObject.class);
+      String name = resultJsonObject.get("viewTemplateName").getAsString();
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_TEMPLATE).
+        canBrowse(false).canSample(true));
+      browseDetailBuilder.addEntity(entity.build());
+      count++;
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+    throws IOException {
+    AribaServices aribaServices = new AribaServices(config);
+    try {
+      accessToken = aribaServices.getAccessToken();
+    } catch (AribaException e) {
+      LOG.error("Error in generating token", e);
+    }
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(io.cdap.plugin.common.ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String template = connectorSpecRequest.getPath();
+    if (template != null) {
+      properties.put(AribaPluginConfig.TEMPLATE_NAME, template);
+    }
+    try {
+      Schema schema = aribaServices.buildOutputSchema(accessToken, template, "prod");
```

Comment:
don't hardcode it as prod, add it in connection config instead, user can create two connections based on type.

Reply:
Done